src/irr/asset/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -231,50 +256,64 @@ layout (set = 3, binding = 5) uniform sampler2D map_d;
 layout (set = 3, binding = 6) uniform sampler2D map_bump;
 #endif //_IRR_FRAG_SET3_BINDINGS_DEFINED_
 
-#include <irr/builtin/glsl/brdf/specular/fresnel/fresnel.glsl>
+#include <irr/builtin/glsl/bsdf/brdf/specular/fresnel/fresnel.glsl>
 
-#ifndef _IRR_FRAG_MAIN_DEFINED_
-#define _IRR_FRAG_MAIN_DEFINED_
+#ifndef _IRR_COMPUTE_LIGHTING_DEFINED_
+#define _IRR_COMPUTE_LIGHTING_DEFINED_
 
-#define Ia 0.1
-void main()
+irr_glsl_BSDFIsotropicParams irr_glsl_computeLighting()
 {
-    vec3 N = normalize(Normal);
+    //would use irr_glsl_calcFragmentShaderSurfaceInteraction(), but then dFd_(ViewPos) would be computed twice
+    irr_glsl_ViewSurfaceInteraction interaction;
+    interaction.V.dir = -ViewPos;
+    interaction.V.dPosdScreen[0] = dFdx(ViewPos);
+    interaction.V.dPosdScreen[1] = dFdy(ViewPos);
+    interaction.N = normalize(Normal);
+
     if ((PC.params.extra&map_bump_MASK) == map_bump_MASK)
     {
         float height = texture(map_bump, UV).x;
-        vec3 dpdx = dFdx(ViewPos);
-        vec3 dpdy = dFdy(ViewPos);
+        vec3 dpdx = interaction.V.dPosdScreen[0];
+        vec3 dpdy = interaction.V.dPosdScreen[1];
         float dhdx = dFdx(height);
         float dhdy = dFdy(height);
         
-        vec3 r1 = cross(dpdy, N);
-        vec3 r2 = cross(N, dpdx);
+        vec3 r1 = cross(dpdy, interaction.N);
+        vec3 r2 = cross(interaction.N, dpdx);
         vec3 surfGrad = (r1*dhdx + r2*dhdy) / dot(dpdx,r1);
-        N = normalize(N - surfGrad);
+        interaction.N = normalize(interaction.N - surfGrad);
     }
-    vec3 L = normalize(-ViewPos);
-    vec3 R = normalize(-reflect(L,N));
-    float NdotL = max(dot(N,L), 0.0);
-    float VdotR = max(dot(L,R), 0.0);
+    return irr_glsl_calcBSDFIsotropicParams(interaction, -ViewPos);
```

Comment:
`computeLighting` is actually supposed to call the `irr_bsdf_cos_eval` function

Reply:
user might want to use parameters got from `irr_glsl_calcBSDFIsotropicParams` (called in `computeLighting`), just like in this case. So im making `out irr_glsl_BSDFIsotropicParams params` parameter for `computeLighting`, ok?

Reply:
> user might want to use parameters

for what exactly?

but irr_glsl_BSDFIsotropicParams gives you BRDF parameters for a particular light interaction, and `computeLighting` will compute the interactions with many lights!

Reply:
>for what exactly?

now it's used to compute alpha (illum models 4,6,7,9)

>computeLighting will compute the interactions with many lights!

oh, right... any ideas?

Reply:
you can return the surface interaction, but not the BRDF params ;)